File: app/src/main/java/com/example/distribution/p007/MainActivity.java
package com.example.distribution.p007;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.TextView;

import java.io.Serializable;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final ArrayList<CompanyDetail> companies = new ArrayList<>();
        CompanyDetail company;
        for(int i= 0;i<10;i++){
            company = new CompanyDetail();
            company.setCompanyName("شرکت"+i);
            company.setCompanyAddress("آدرس شرکت"+i);
            company.setCompanyId(i);
            companies.add(company);

            Log.v("companies",companies.get(i).getCompanyName());
        }

        MainAdapter adapter = new MainAdapter(this,0,companies);
        ListView list = (ListView) findViewById(R.id.list_view);
        list.setAdapter(adapter);
        //----------------------- company item selection --------------
        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
            Intent intent = new Intent(MainActivity.this,CompanyActivity.class);
            intent.putExtra("company", companies.get(position));
            startActivity(intent);

            }
        });

    }
}

File: app/src/main/java/com/example/distribution/p007/MainActivity.kt
package com.example.distribution.p007

import android.os.Build
import android.os.Bundle
import android.support.annotation.RequiresApi
import android.support.v7.app.AppCompatActivity
import android.view.View

class MainActivity : AppCompatActivity() {


    val TAG = "MainFragment"
    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        window.decorView.layoutDirection = View.LAYOUT_DIRECTION_RTL
        //---------------------main fragment----------------------------
        val companyListMain = CompanyListMain_Fr()
        fragmentManager.beginTransaction().add(R.id.fragment_container,companyListMain,TAG).commit()

    }

    override fun onBackPressed() {
        var myFragment = fragmentManager.findFragmentById(R.id.fragment_container)
        if (myFragment != null && (myFragment is CompanyListMain_Fr || myFragment is GoodsProfile_Fr) && myFragment.isVisible) {
            super.onBackPressed()
        }else{
            val companyListMain = CompanyListMain_Fr()
            fragmentManager.beginTransaction().add(R.id.fragment_container,companyListMain).commit()
        }

    }


}
